features/include/pcl/features/flare.h

```diff
@@ -0,0 +1,300 @@
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*
+*
+*/
+
+#ifndef PCL_FLARE_H_
+#define PCL_FLARE_H_
+
+#include <pcl/point_types.h>
+#include <pcl/features/feature.h>
+#include <pcl/features/normal_3d_omp.h>
+
+
+namespace pcl
+{
+  /** \brief FLARELocalReferenceFrameEstimation implements the Fast LocAl Reference framE algorithm
+  * for local reference frame estimation as described here:
+  *
+  *  - A. Petrelli, L. Di Stefano,
+  *    "A repeatable and efficient canonical reference for surface matching",
+  *    3DimPVT, 2012
+  *
+  * FLARE algorithm is deployed in ReLOC algorithm proposed in:
+  * 
+  * Petrelli A., Di Stefano L., "Pairwise registration by local orientation cues", Computer Graphics Forum, 2015.
+  *
+  * \author Alioscia Petrelli
+  * \ingroup features
+  */
+  template<typename PointInT, typename PointNT, typename PointOutT = ReferenceFrame, typename SignedDistanceT = float>
+  class FLARELocalReferenceFrameEstimation : public FeatureFromNormals<PointInT, PointNT, PointOutT>
+  {
+  protected:
+    using Feature<PointInT, PointOutT>::feature_name_;
+    using Feature<PointInT, PointOutT>::input_;
+    using Feature<PointInT, PointOutT>::indices_;
+    using Feature<PointInT, PointOutT>::surface_;
+    using Feature<PointInT, PointOutT>::tree_;
+    using Feature<PointInT, PointOutT>::search_parameter_;
+    using FeatureFromNormals<PointInT, PointNT, PointOutT>::normals_;
+    using Feature<PointInT, PointOutT>::fake_surface_;
+    using Feature<PointInT, PointOutT>::getClassName;
+
+    typedef typename Feature<PointInT, PointOutT>::PointCloudIn PointCloudIn;
+    typedef typename Feature<PointInT, PointOutT>::PointCloudOut PointCloudOut;
+
+    typedef typename pcl::PointCloud<SignedDistanceT> PointCloudSignedDistance;
+    typedef typename PointCloudSignedDistance::Ptr PointCloudSignedDistancePtr;
+
+    typedef typename PointCloudIn::ConstPtr PointCloudInConstPtr;
+
+    typedef boost::shared_ptr<FLARELocalReferenceFrameEstimation<PointInT, PointNT, PointOutT> > Ptr;
+    typedef boost::shared_ptr<const FLARELocalReferenceFrameEstimation<PointInT, PointNT, PointOutT> > ConstPtr;
+
+    typedef typename pcl::search::Search<PointInT>::Ptr KdTreePtr;
+
+
+
+  public:
+
+    /** \brief Constructor. */
+    FLARELocalReferenceFrameEstimation () :
+        tangent_radius_ (0.0f),
+          margin_thresh_ (0.85f),
+          min_neighbors_for_normal_axis_(6),
+          min_neighbors_for_tangent_axis_(6),
+          sampled_surface_(), 
+          sampled_tree_(),
+          fake_sampled_surface_(false)
+        {
+          feature_name_ = "FLARELocalReferenceFrameEstimation";
+        }
+
+        /** \brief Empty destructor */
```

Comment:
I now realize you declared this because of your other virtual methods. If we agree later on that they don't need to be virtual, then this can be deleted.